src/fundamental/chid-fundamental.h

```diff
@@ -0,0 +1,29 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#pragma once
+
+#include "string-util-fundamental.h"
+
+typedef enum ChidSmbiosFields {
+        CHID_SMBIOS_MANUFACTURER,
+        CHID_SMBIOS_FAMILY,
+        CHID_SMBIOS_PRODUCT_NAME,
+        CHID_SMBIOS_PRODUCT_SKU,
+        CHID_SMBIOS_BASEBOARD_MANUFACTURER,
+        CHID_SMBIOS_BASEBOARD_PRODUCT,
+        CHID_SMBIOS_COUNT,
```

Comment:
in our codebase we sytematically call these special counter enum values `_FOO_BAR_MAX` for an enum type FooBar.

hence, please rename this last entry to `_CHID_SMBIOS_FIELDS_MAX`. (yes, please use the full type nae as prefix before the _MAX, just to follow our usual naming style. It's actually find if you don't repeat the full type name for the other enum values.